src/nvim/os/signal.c

```diff
@@ -212,10 +212,14 @@ static void on_signal(SignalWatcher *handle, int signum, void *data)
     // Ignore
     break;
 #endif
-  case SIGTERM:
 #ifdef SIGTSTP
   case SIGTSTP:
+    if (p_awa) {
+      autowrite_all();
```

Comment:
Should we drop this for now, given https://github.com/neovim/neovim/pull/33258#issuecomment-2772231011 ? May need to adjust a test.

Reply:
Yeah I think it couldn't be merged until we solved this problem, and although i kinda figured out why it doesn't work it seems to be really tricky to fix. Maybe need to turn it into a draft for now.

Reply:
Not enough to remove this line, we need also remove all the code that registers (and the corresponding cleanup code) `SIGTSTOP` handler, or it will still be a blink. As long as the handler exists, we need to suspend by ourselves.

Reply:
> Not enough to remove this line, we need also remove all the code that registers (and the corresponding cleanup code) `SIGTSTOP` handler, or it will still be a blink. As long as the handler exists, we need to suspend by ourselves.

It's basically reverting most of https://github.com/neovim/neovim/commit/4dabeff308222307ede8e74a2bd341713a7f7d81 (since it mainly adding handlers for `SIGSTOP`, along with some other stuffs)

Reply:
Sure, let's revert all `sigstp`-related changes from that commit. But the changes for the other signals can stay.

Reply:
> Sure, let's revert all `sigstp`-related changes from that commit. But the changes for the other signals can stay.

I checked that commit carefully and confirmed that it's just all about adding handlers to `SIGTSTP` (and adding corresponding tests and documentation), so if removing the code is preferred solution, then it's just revert this commit and all done

Reply:
You forgot this:
```c
  if (p_awa && signum != SIGTERM) {
    autowrite_all();
  } 
```

Reply:
> You forgot this:
> 
> ```c
>   if (p_awa && signum != SIGTERM) {
>     autowrite_all();
>   } 
> ```

oh yeah. but others are almost all about `SIGTSTP` and shall be removed

Reply:
Oh wait... I might find out a nice solution that can preserve this feature and also pause `nvim` normally. I'm going to run tests now.